modules/effects/arrow-effects-rx2/src/main/kotlin/arrow/effects/ObservableK.kt

```diff
@@ -27,6 +33,14 @@ data class ObservableK<A>(val observable: Observable<A>) : ObservableKOf<A>, Obs
   fun <B> flatMap(f: (A) -> ObservableKOf<B>): ObservableK<B> =
     observable.flatMap { f(it).fix().observable }.k()
 
+  fun <B> bracketCase(use: (A) -> ObservableKOf<B>, release: (A, ExitCase<Throwable>) -> ObservableKOf<Unit>): ObservableK<B> =
+    flatMap { a ->
+      use(a).fix().observable
+        .doOnNext { release(a, ExitCase.Completed) }
```

Comment:
doOnComplete

Reply:
Fixed. Following [this](https://github.com/ReactiveX/RxJava/issues/4456) and [this](https://github.com/ReactiveX/RxJava/pull/4458) looks like `Observable.onDispose()` would be the equivalent to `Flowable.onCancel()`, so I'm notifying the `Cancelled` ExitCase when the observer is disposed.

Added a test to cover the cancellation.